### PR TITLE
Removed ddtrace.propagation.utils module

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
The deprecated `ddtrace.propagation.utils` module was removed.

### CODE REVIEW

File: ddtrace/propagation/utils.py [hunk 1]
```diff
@@ -1,31 +0,0 @@
-from typing import Optional
```
Comment (brettlangdon) [inline ddtrace/propagation/utils.py:1]: do we need to keep these methods here with a deprecation warning?

Comment (Yun-Kim) [reply ddtrace/propagation/utils.py:1]: This PR is meant for v1.0, but I'll make a separate PR for deprecating first 😄

Comment (Yun-Kim) [reply ddtrace/propagation/utils.py:1]: This PR is meant for v1.0, but I'll make a separate PR for deprecating first 😄

**Comment by Yun-Kim** [comment]
This PR is blocked until #2651 (into which deprecation changes were cherry-picked) is merged as this simply removes the `ddtrace.propagation.utils` module.

**Review by brettlangdon — APPROVED** [review]

**Comment by Yun-Kim** [comment]
Closing after offline discussion to include the `propagation.utils` module into a `ddtrace.tracing` module, therefore making this dependent on a future and much bigger change relating to `trace_utils`.

**Pull request closed without merging**